.github/workflows/test.yml

```diff
@@ -23,11 +23,21 @@ jobs:
     steps:
       - uses: actions/checkout@v3
         with:
-          submodules: recursive
+          submodules: false
+
+      - uses: actions/cache@v3
+        id: foundry-cache
+        with:
+          path: tests/lib
+          key: cache-foundry-${{ hashFiles('.gitmodules') }}
```

Comment:
looks like this may be causing issues in the checkout post-job cleanup, but the tests still succeed 🤷‍♂️ 

```
/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
fatal: not a git repository: tests/lib/forge-std/../../../.git/modules/tests/lib/forge-std
Warning: The process '/usr/bin/git' failed with exit code 128
```

Reply:
my understanding is that the cache key needs to be unique for the specific snapshot of submodules.

the .gitmodules file itself doesn't include the refs hash for the submodules.  so even if existing submodules are updated to a newer version, the cache key will be the same, which will load the old submodules from the cache.

is there anything i'm missing here?

Reply:
yes this key could lead to a stale cache if the intent is to fetch the latest version of each dependency, since we don't have a lockfile equivalent AFAIK. This is just to experiment with caching and guaranteeing hits for now -- we could use a shorter lived key including for instance the date

Reply:
i think the ci cache is nontrivial and deserves its own pr.  can we move this ci update to a separate pr?